gtsfm/densify/mvs_patchmatchnet.py

```diff
@@ -311,4 +355,82 @@ def filter_depth(
 
         dense_points = np.concatenate(vertices, axis=0)
         dense_point_colors = np.concatenate(vertex_colors, axis=0)
-        return dense_points, dense_point_colors
+
+        # compute and collect metrics during filtering points from depth maps
+        filter_metrics = []
+        # compute the proportion of valid pixels in the geometric masks among all reference views
+        filter_metrics.append(GtsfmMetric(name="geometric_mask_valid_ratios", data=geo_mask_ratios))
+        # compute the proportion of valid pixels in the confidence masks among all reference views
+        filter_metrics.append(GtsfmMetric(name="confidence_mask_valid_ratios", data=conf_mask_ratios))
+        # compute the proportion of valid pixels in the joint masks among all reference views
+        filter_metrics.append(GtsfmMetric(name="joint_mask_valid_ratios", data=joint_mask_ratios))
+        filter_metrics.append(GtsfmMetric(name="reprojection_errors", data=reprojection_errors, store_full_data=False))
+
+        return dense_points, dense_point_colors, GtsfmMetricsGroup(name="filter metrics", metrics=filter_metrics)
+
+    def compute_filtered_reprojection_error(
```

Comment:
does this need to be part of the class, or can it be a free function? prefer functional style.

Reply:
marking as resolved, thanks!